src/kernel/imfilter_cuda.jl

```diff
@@ -0,0 +1,27 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+const array_kernel(::CUDAMethod, array) = CuArray{Float32}(array)
+
+const view_kernel(::CUDAMethod, array, I) = Array(array[I])
+
+function imfilter_kernel(::CUDAMethod, img, krn)
+  # retrieve basic info
+  N = ndims(img)
+  T = eltype(img)
+
+  # pad kernel to common size with image
+  padkrn = CUDA.zeros(Float32, size(img))
```

Comment:
@ClaroHenrique I thought we were allocating this array once outside the main loop and just copying the kernel in the top left corner. Can you please explain? We are taking too much time to finish this up.

Reply:
We tried to allocate a padded kernel outside the loop in this [PR](https://github.com/JuliaEarth/ImageQuilting.jl/pull/45/files), making the code more complicated.
But as it turns out, allocating space to the pad kernel in GPU with zeros and copying only the kernel in CPU was enough to optimize the code.